### DISCUSSION TITLE
Method to add one get parameter to all pages on the site

### DISCUSSION THREAD

**Discussion by Rikijs** [symfony/symfony#49534] [Q&A] [answered]
Hello!

In order to track opened pages (allow only one page open) in multiple browser tabs i need a way to store `window-id`.
I figure - it should be possible to add get parameter with `window-id` to **every** _password protected_ page.
I would like to _avoid adding it manually_ for every action in all controllers.
Afterwards i plan to check get parameter with `window-id` on each _Kernel request_ in `RequestSubscriber`.
Is there a method to _add one get parameter_ to **all** _restricted pages_ on the site?

Thank you!

**Comment by ghost** [comment, marked as answer]
I think this will be hard to achieve on the backend side. To add a GET parameter to every URL generated by Symfony you can decorate UrlGenerator but there are other problems to solve:
- user types a protected URL by hand in multiple tabs
- you will not know whether a user opened a link it in the same tab or in a new tab (CTRL+left mouse button). As a result you will have 2 tabs with the same window-id query parameter.
- how will you know on the backend side that a user closed a tab and now can open another protected tab?
- remember that relying on the query parameter existence to enforce security is not the best idea because parameter can be removed (removal however can be verified in request listener) or changed by a malicious user (he can take it from already opened tab).

I think solution to you problem should be solved on the frontend side with JavaScript and Tabs API.
